tests/integration/sentinel/transitions/create-user-for-contacts.spec.js

```diff
@@ -0,0 +1,533 @@
+    await utils.updateSettings(getSettings(), 'sentinel');
+    await utils.createUsers([ORIGINAL_USER]);
+    newUsers.push(ORIGINAL_USER.username);
+    await utils.saveDoc(NEW_PERSON);
+    const originalContact = await utils.getDoc(ORIGINAL_PERSON._id);
+    originalContact.user_for_contact = {
+      replace: {
+        [ORIGINAL_USER.username]: {
+          replacement_contact_id: NEW_PERSON._id,
+          status: 'PENDING'
+        }
+      },
+    };
+    await utils.saveDoc(originalContact);
+    await sentinelUtils.waitForSentinel(originalContact._id);
+    const { transitions } = await sentinelUtils.getInfoDoc(originalContact._id);
+
+    assert.isUndefined(transitions.create_user_for_contacts);
+
+    // Original contact not updated
+    const originalPerson = await utils.getDoc(ORIGINAL_PERSON._id);
+    assert.isUndefined(originalPerson.errors);
+    // New user not created
+    const newUserSettings = await utils.getUserSettings({ contactId: NEW_PERSON._id });
+    assert.isEmpty(newUserSettings);
+  });
+
+  it('does not replace user when the contact is not being replaced', async () => {
+    await utils.updateSettings(getSettings(), 'sentinel');
+    await utils.createUsers([ORIGINAL_USER]);
+    newUsers.push(ORIGINAL_USER.username);
+    await utils.saveDoc(NEW_PERSON);
+    const originalContact = await utils.getDoc(ORIGINAL_PERSON._id);
+    originalContact.name = 'Updated Person';
+    await utils.saveDoc(originalContact);
+    await sentinelUtils.waitForSentinel(originalContact._id);
+    const { transitions } = await sentinelUtils.getInfoDoc(originalContact._id);
+
+    assert.isUndefined(transitions.create_user_for_contacts);
+
+    // Original contact not updated
+    const originalPerson = await utils.getDoc(ORIGINAL_PERSON._id);
+    assert.isUndefined(originalPerson.errors);
+    // New user not created
+    const newUserSettings = await utils.getUserSettings({ contactId: NEW_PERSON._id });
+    assert.isEmpty(newUserSettings);
+  });
+
+  it('does not replace user when the contact being replaced is not a person', async () => {
+    await utils.updateSettings(getSettings(), 'sentinel');
+    await utils.createUsers([ORIGINAL_USER]);
+    newUsers.push(ORIGINAL_USER.username);
+    await utils.saveDoc(NEW_PERSON);
+    const clinic = await utils.getDoc(CLINIC._id);
+    clinic.user_for_contact = { replace: { by: NEW_PERSON._id, status: 'READY' } };
+    await utils.saveDoc(clinic);
+    await sentinelUtils.waitForSentinel(clinic._id);
+    const { transitions } = await sentinelUtils.getInfoDoc(clinic._id);
+
+    assert.isUndefined(transitions.create_user_for_contacts);
+
+    // Original contact not updated
+    const originalPerson = await utils.getDoc(ORIGINAL_PERSON._id);
+    assert.isUndefined(originalPerson.errors);
+    // New user not created
+    const newUserSettings = await utils.getUserSettings({ contactId: NEW_PERSON._id });
+    assert.isEmpty(newUserSettings);
```

Comment:
I think that reparenting after an error can become very confusing, especially from an analytics standpoint. Down the line, when data is analyzed, it will find a contact, that was never associated to a user, submit reports.

Reply:
Made some changes in commit [`2a35308` (#7709)](https://github.com/medic/cht-core/pull/7709/commits/2a3530829955a267dc322095ad68b6ecf25b4665) to prevent reparenting new reports and a test covering it